torch_uncertainty/routines/pixel_regression.py

```diff
@@ -174,39 +173,47 @@ def training_step(self, batch: tuple[Tensor, Tensor], batch_idx: int) -> STEP_OU
         if self.one_dim_depth:
             target = target.unsqueeze(1)
 
-        dists = self.model(inputs)
-        out_shape = dist_size(dists)[-2:] if self.probabilistic else dists.shape[-2:]
+        out = self.model(inputs)
+        out_shape = out[next(iter(out))].shape[-2:] if self.probabilistic else out.shape[-2:]
         target = F.resize(target, out_shape, interpolation=F.InterpolationMode.NEAREST)
-        padding_mask = torch.isnan(target)
+        target = rearrange(target, "b c h w -> b h w c")
+        padding_mask = torch.isnan(target).any(dim=-1)
         if self.probabilistic:
+            dist_params = {k: rearrange(v, "b c h w -> b h w c") for k, v in out.items()}
+            dists = Independent(get_dist_class(self.dist_family)(**dist_params), 1)
```

Comment:
Add a comment to explain why we need Independent (we're not all experts on torch.distributions :smile: )